bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java

```diff
@@ -124,61 +157,125 @@ public synchronized void sendPacket(byte[] packet) {
+    private void writePacket(byte[] packet) {
+        if (outputStream == null) {
+            onConnectionLost();
+            return;
+        }
+
+        try {
+            outputStream.write(packet);
+            outputStream.flush();
+        } catch (IOException e) {
+            onConnectionLost();
+        }
     }
 
-    @Override
-    public void dispose() {
-        if (serialPort != null) {
-            serialPort.removeEventListener();
-            serialPort.close();
-            serialPort = null;
+    protected void onConnectionLost() {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                "A network communication error occurred.");
+        disconnect();
+        startReconnectionHandler();
+    }
+
+    protected abstract void connect();
+
+    protected void disconnect() {
+        listenerStopped = true;
+
+        try {
+            if (outputStream != null) {
+                outputStream.close();
+            }
+        } catch (IOException e) {
+            logger.error("Error while closing output stream", e);
+        }
+
+        try {
+            if (inputStream != null) {
+                inputStream.close();
+            }
+        } catch (IOException e) {
+            logger.error("Error while closing input stream", e);
         }
     }
 
-    @Override
-    public void serialEvent(SerialPortEvent event) {
-        logger.debug("Serial port event triggered");
-
-        if (event.getEventType() == SerialPortEvent.DATA_AVAILABLE) {
-            try {
-                byte[] packet;
-                while ((packet = inputStream.readPacket()) != null) {
-                    byte address = packet[2];
-
-                    VelbusPacketListener packetListener = packetListeners.get(address);
-                    if (packetListener != null) {
-                        packetListener.onPacketReceived(packet);
-                    } else if (defaultPacketListener != null) {
-                        defaultPacketListener.onPacketReceived(packet);
-                    }
+    public void startReconnectionHandler() {
+        if (reconnectionHandler == null || reconnectionHandler.isCancelled()) {
+            Object reconnectionIntervalObject = getConfig().get(RECONNECTION_INTERVAL);
+            if (reconnectionIntervalObject != null) {
+                long reconnectionInterval = ((BigDecimal) reconnectionIntervalObject).longValue();
+
+                if (reconnectionInterval > 0) {
+                    reconnectionHandler = scheduler.scheduleWithFixedDelay(new Runnable() {
+
+                        @Override
+                        public void run() {
+                            try {
+                                connect();
+                                if (reconnectionHandler != null) {
+                                    reconnectionHandler.cancel(false);
+                                }
+                            } catch (Exception e) {
```

Comment:
Modified in commit 780323c: replaced the exception catching by a boolean check on "connect()". In case of problems during the connection, the "connect()" operation is setting the thing to offline through "onConnectionLost()".